File: samples_generated_code/samples_2/phi4/baseline_code_java/code_row_80.java
import java.util.HashMap;
import java.util.Scanner;

public class SimpleLoginPage {
    private static final HashMap<String, String> users = new HashMap<>();
    private static final Scanner scanner = new Scanner(System.in);

    public static void main(String[] args) {
        // Sample user for demonstration purposes
        users.put("user@example.com", "password123");

        System.out.println("Welcome to Simple Login Page");
        
        while (true) {
            System.out.print("Enter email: ");
            String email = scanner.nextLine();
            System.out.print("Enter password: ");
            String password = scanner.nextLine();

            if (authenticate(email, password)) {
                showMenu(email);
            } else {
                System.out.println("Authentication failed. Please try again.");
            }
        }
    }

    private static boolean authenticate(String email, String password) {
        return users.containsKey(email) && users.get(email).equals(password);
    }

    private static void showMenu(String currentEmail) {
        while (true) {
            System.out.println("\nLogged in as: " + currentEmail);
            System.out.println("1. Change Email");
            System.out.println("2. Logout");

            System.out.print("Choose an option: ");
            String choice = scanner.nextLine();

            if ("1".equals(choice)) {
                changeEmail(currentEmail);
            } else if ("2".equals(choice)) {
                break;
            } else {
                System.out.println("Invalid option, please try again.");
            }
        }
    }

    private static void changeEmail(String currentEmail) {
        while (true) {
            System.out.print("Enter old email: ");
            String oldEmail = scanner.nextLine();
            if (!oldEmail.equals(currentEmail)) {
                System.out.println("Old email does not match. Please try again.");
                continue;
            }

            System.out.print("Enter new email: ");
            String newEmail = scanner.nextLine();

            if (users.containsKey(newEmail)) {
                System.out.println("New email is already taken. Please try another one.");
                continue;
            }
            
            System.out.print("Confirm password for old email: ");
            String confirmPassword = scanner.nextLine();
            
            if (!confirmPassword.equals(users.get(currentEmail))) {
                System.out.println("Incorrect confirm password. Please try again.");
                continue;
            }

            users.remove(oldEmail);
            users.put(newEmail, users.get(oldEmail));
            System.out.println("Email changed successfully!");

            break;
        }
    }
}

File: samples_generated_code/samples_2/athene/baseline_code_c/code_row_80.c
#include <stdio.h>
#include <string.h>

#define MAX_EMAIL_LEN 100
#define MAX_PASSWORD_LEN 50

typedef struct {
    char email[MAX_EMAIL_LEN];
    char password[MAX_PASSWORD_LEN];
} User;

int authenticate(const User *user, const char *email, const char *password) {
    return strcmp(user->email, email) == 0 && strcmp(user->password, password) == 0;
}

void changeEmail(User *user) {
    char oldEmail[MAX_EMAIL_LEN], newEmail[MAX_EMAIL_LEN], confirmPassword[MAX_PASSWORD_LEN];

    printf("Enter your old email: ");
    scanf("%99s", oldEmail);
    printf("Enter your new email: ");
    scanf("%99s", newEmail);
    printf("Confirm your password: ");
    scanf("%49s", confirmPassword);

    if (authenticate(user, oldEmail, confirmPassword)) {
        strcpy(user->email, newEmail);
        printf("Email changed successfully.\n");
    } else {
        printf("Authentication failed. Please check your email and password.\n");
    }
}

int main() {
    User user = {"example@example.com", "password123"};

    char currentEmail[MAX_EMAIL_LEN];
    printf("Enter your current email to log in: ");
    scanf("%99s", currentEmail);

    if (strcmp(user.email, currentEmail) == 0) {
        changeEmail(&user);
    } else {
        printf("User not found. Please try again.\n");
    }

    return 0;
}